example/CpImport/src/main/java/com/mariadb/columnstore/api/examples/CpImport.java

```diff
@@ -0,0 +1,260 @@
+		System.out.println("Invalid count: " + summary.getInvalidCount());
+		return;
+	}
+
+	/**
+	 * Imports the csv-encoded file and parses dates according to dateFormat and
+	 * dateTimeFormat.
+	 * 
+	 * @param targetTable,
+	 *            target table
+	 * @param importFile,
+	 *            csv-encoded file to import
+	 * @param dateFormat,
+	 *            date format to interpret based on Java's SimpleDateFormat notation
+	 * @param dateTimeFormat,
+	 *            dateTime format to interpret based on Java's SimpleDateFormat
+	 *            notation
+	 */
+	public void importFile(String targetTable, File importFile, String dateFormat, String dateTimeFormat) {
+		ColumnStoreBulkInsert b = d.createBulkInsert(DB_NAME, targetTable, (short) 0, 0);
+
+		// get the row count of targetTable
+		ColumnStoreSystemCatalog catalog = d.getSystemCatalog();
+		ColumnStoreSystemCatalogTable table = catalog.getTable(DB_NAME, targetTable);
+		int targetTableColumnCount = table.getColumnCount();
+
+		// insert row by row into targetTable from importFile
+		try (BufferedReader br = new BufferedReader(new FileReader(importFile))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] columns = line.split(delimiter);
+				for (int i = 0; i < columns.length; i++) {
+					if (i < targetTableColumnCount) {
+						if (table.getColumn(i).getType().equals(columnstore_data_types_t.DATA_TYPE_DATE)) {
+							b.setColumn(i, parseDate(columns[i], dateFormat));
+						} else if (table.getColumn(i).getType().equals(columnstore_data_types_t.DATA_TYPE_DATETIME)) {
+							b.setColumn(i, parseDateTime(columns[i], dateTimeFormat));
+						} else {
+							b.setColumn(i, columns[i]);
+						}
+					}
+				}
+				b.writeRow();
+			}
+			b.commit();
+		} catch (FileNotFoundException e) {
+			System.err.println("import-file doesn't exist");
+			System.exit(3);
+		} catch (IOException e) {
+			System.err.println("error while reading / parsing the input-file: " + e.getMessage());
+		} catch (Exception e) {
+			System.err.println("error during mcsapi write operations: " + e.getMessage());
+		}
+
+		// print a short summary of the insertion process
+		ColumnStoreSummary summary = b.getSummary();
+		System.out.println("Execution time: " + summary.getExecutionTime());
+		System.out.println("Rows inserted: " + summary.getRowsInsertedCount());
+		System.out.println("Truncation count: " + summary.getTruncationCount());
+		System.out.println("Saturated count: " + summary.getSaturatedCount());
+		System.out.println("Invalid count: " + summary.getInvalidCount());
+		return;
+	}
+
+	/**
+	 * Parses the date to a MariaDB supported format.
+	 * 
+	 * @param date,
+	 *            input to import
+	 * @param dateFormat,
+	 *            of input in SimpleDateFormat notation
+	 * @return parsed date in MariaDB format, or null in case of error.
+	 */
+	private String parseDate(String date, String dateFormat) {
+		SimpleDateFormat inputParser = new SimpleDateFormat(dateFormat);
```

Comment:
I believe it's still true that SimpleDateFormat is still considered somewhat expensive to construct each time so moving it to a class member variable and reusing might be faster. Same for method below.